### PR TITLE
change naming convension from intf to if, _i to if_i, and tb_top to tb

### PR DESCRIPTION
**Pull request by silabs-krdosvik**
Changing the guidelines as we have not followed them.

### CODE REVIEW

File: docs/CodingStyleGuidelines.md
Comment (silabs-krdosvik) [inline docs/CodingStyleGuidelines.md:104]: I believe we have used tb and not tb_top, so I changed this as well. Another alternative is to change the module name. Or maybe I have misunderstood what we are referring to here and there should be no change.

File: docs/CodingStyleGuidelines.md
Comment (silabs-krdosvik) [inline docs/CodingStyleGuidelines.md:144]: I see that we have not been consistent in the way we have named our interface instances. Some interface instances have the postfix _i (actually if_i), and some have _if. As the guidelines originally state that we sould use _i, I changed this to _if_i, and not only _if. To be consistent we will need some change in the code base, but should we go for _if_i, _if or only _i?

File: docs/CodingStyleGuidelines.md
Comment (silabs-krdosvik) [inline docs/CodingStyleGuidelines.md:743]: I didnt see any use of the macro cmn_get_intf when running the grep command on the core-v-verif repo, so I assume it is okey to just change the macro name? Not sure though, as the macro is probably defined somewhere?

File: docs/CodingStyleGuidelines.md
Comment (silabs-krdosvik) [inline docs/CodingStyleGuidelines.md:756]: If we decide to use the _if_i or _if as a standard for interface instances, we should probably also change _vi to _if_vi or _vif for the virtual interfaces as well, or what do you think? 

(I see I have misspelled virtual in the comment... When writing viritual, I mean to "say" virtual)

File: docs/CodingStyleGuidelines.md
Comment (MikeOpenHWGroup) [reply docs/CodingStyleGuidelines.md:104]: Good catch

File: docs/CodingStyleGuidelines.md
Comment (MikeOpenHWGroup) [reply docs/CodingStyleGuidelines.md:144]: I would prefer `_if`.  Typically you will see `_i` to indicate an input signal to a module, so using it to designate an interface may confuse some people.

File: docs/CodingStyleGuidelines.md
Comment (MikeOpenHWGroup) [reply docs/CodingStyleGuidelines.md:743]: We have not really discussed these macros.  The names used in the guidelines are as they were from the original contribution, and they really should be reviewed.  For now, the change you have made is OK.

File: docs/CodingStyleGuidelines.md
Comment (silabs-robin) [reply docs/CodingStyleGuidelines.md:144]: Seems we use `..._i` for module instances. Sometimes also `u_...`.

Seems we use `..._if` for both interface types and interface instances.

File: docs/CodingStyleGuidelines.md
Comment (silabs-robin) [reply docs/CodingStyleGuidelines.md:756]: We do use `..._vif`.

File: docs/CodingStyleGuidelines.md
Comment (silabs-robin) [inline docs/CodingStyleGuidelines.md:98]: Should the target be `master` or `cv32e40s/dev`?

**Review by silabs-mateilga — COMMENTED** [review]
My personal opinion is _if for both interface instances and types, and _vif for virtual interfaces.

File: docs/CodingStyleGuidelines.md
Comment (MikeOpenHWGroup) [reply docs/CodingStyleGuidelines.md:98]: Good point @silabs-robin.  Since this is a documentation item, which affects all projects using core-v-verif, it should probably be `master`.  Having said that, `e40s/dev` will eventually bubble up to master, so I am not strongly opinionated.

Comment (silabs-robin) [reply docs/CodingStyleGuidelines.md:98]: cv32e40s/dev is fine 👍

**Comment by silabs-hfegran** [comment]
Oh hey there, another coding standard discussion, count me in!

Here is my _personal_ preference, firstly the directly relevant ones for this pr, the others are for completeness: 
* Suffix all interfaces with `_if`
* Suffix all virtual interfaces with `_vif`
* Suffix all types with _t (additively),  thus an interface type to me would make sense to have as `somename_if_t`.
* Suffix all enum types with `_e` (additively)
* Suffix inputs, inouts and outputs with `_in`, `_inout`, `_out`
* Suffix constraints with `_c`
* Suffix internal variables with `_i` (I guess this can be somewhat controversial, as many prefer this as input)
* Suffix current state, next state of regs with `_q`, `_n`
* Prefix instances with `u_`, prevents confusion with multiple naming schemes use of `_i`
* Prefix classes with `c_`
* Prefix non-parameter constants with `const_`

**Comment by silabs-mateilga** [comment]
As usual I mostly agree with Henrik, but not completely. 
For internal variables I prefer to have no suffix. If we use `_in`, `_inout` and `_out` for all interfacing signals, then there is no need to mark internal variables. Also, you avoid the confusion with `_i` being commonly used as input.

**Comment by silabs-robin** [comment]
Did the discussion converge, and were the files updated accordingly?

**Comment by silabs-krdosvik** [comment]
> Did the discussion converge, and were the files updated accordingly?
**Discussion:** As far as I can see, Marton and Henrik have different opinions of how to name interface types, if and if_t, respectively (but I think Marton might have change his mind and says if_t is good). Also there are different opinions about the _in, _out, and _inout suffix.
**Files updated:** I put this task on hold as other tasks are more critical atm.

**Comment by silabs-mateilga** [comment]
Yes, if_t for interface types is a good option.

**Review by MikeOpenHWGroup — APPROVED** [review]

**Comment by MikeOpenHWGroup** [comment]
@silabs-krdosvik, @silabs-hfegran, @silabs-mateilga and @silabs-robin:

I have approved this pull-request, but will not merge it.  I _think_ we are all mostly agreed on the recommended updates to the coding style and AFAICT the only outstanding item for this PR is to rename interface types as `if_t`.  I will leave it to the 4 of you to decide whether we should gate this PR on that item or not.

Your participation on this is very much appreciated.

**Comment by silabs-krdosvik** [comment]
> @silabs-krdosvik, @silabs-hfegran, @silabs-mateilga and @silabs-robin:
> 
> I have approved this pull-request, but will not merge it. I _think_ we are all mostly agreed on the recommended updates to the coding style and AFAICT the only outstanding item for this PR is to rename interface types as `if_t`. I will leave it to the 4 of you to decide whether we should gate this PR on that item or not.
> 
> Your participation on this is very much appreciated.

Thanks Mike! 

I dont think we should merge it yet. As it stands now, formal verification works fine, but not simulation.

**Comment by MikeOpenHWGroup** [comment]
> I dont think we should merge it yet. As it stands now, formal verification works fine, but not simulation.

Fair enough.  I have placed a `DO NOT MERGE` label on this pull-request.  Please remove it when you think it is ready.

**Comment by silabs-krdosvik** [comment]
> As usual I mostly agree with Henrik, but not completely. For internal variables I prefer to have no suffix. If we use `_in`, `_inout` and `_out` for all interfacing signals, then there is no need to mark internal variables. Also, you avoid the confusion with `_i` being commonly used as input.

I agree! Also, theoretically, it is possible for a signal in an interface to be input to one module and output of another module.

**Comment by silabs-robin** [comment]
I think this is a nice cleanup. Is it ready to merge?

Does it pass ci_check?
Formal is runnable?
All discussions reached consensus?

**Comment by silabs-hfegran** [comment]
> Does it pass ci_check?
Formal is runnable?

^ Status on this?

> All discussions reached consensus?

I am fine with the discussion part of this, for the sake of reviews I would prefer to keep documentation and code updates separate in the future, but won't block this PR on that. There are however merge conflicts here that needs to be resolved before I can merge.

**Comment by silabs-krdosvik** [comment]
> I think this is a nice cleanup. Is it ready to merge?
> 
> Does it pass ci_check? Formal is runnable? All discussions reached consensus?

The interface type part is not yet finished, will let you know when it is :-)

**Review by MikeOpenHWGroup — APPROVED** [review]
Thanks @silabs-krdosvik!

**Comment by silabs-hfegran** [comment]
@silabs-krdosvik What is the status on this PR?

**Comment by silabs-krdosvik** [comment]
> @silabs-krdosvik What is the status on this PR?

Obs.. it is still on hold, but i guess Ill just finish it now.

**Comment by silabs-krdosvik** [comment]
I think I have included all changes now :-)

File: cv32e40s/env/uvme/uvma_cv32e40s_core_cntrl_agent.sv
Comment (silabs-robin) [inline cv32e40s/env/uvme/uvma_cv32e40s_core_cntrl_agent.sv:1]: Does ci_check still pass, and is formal still runnable?

File: docs/CodingStyleGuidelines.md
Comment (silabs-robin) [inline docs/CodingStyleGuidelines.md:144]: This does not correspond with all the changes in the source files?

**Review by silabs-robin — APPROVED** [review]
I have those two remarks above, but I still want to give an approve already, because the changes look good and I expect that it is somewhat expensive to maintain changes on a long-lived PR than to fix potential details later. (I could be wrong about this.)

Should hold of merging until more people approve.

**Comment by silabs-krdosvik** [comment]
For the moment, the PR is passing ci check and formal. I think I have included most of the necessary changes now, and it is indeed expensive to maintain.

**Review by silabs-hfegran — APPROVED** [review]

**Pull request merged**
